Add GitHub Actions for Snap build and push

As title.

@dewyatt This is the Ubutnu 18.04 log from Github Action worker:
```
CMake Warning at cmake/Modules/Findcmocka.cmake:44 (find_package):
  By not providing "Findcmocka.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "cmocka", but
  CMake did not find one.

  Could not find a package configuration file provided by "cmocka" with any
  of the following names:

    cmockaConfig.cmake
    cmocka-config.cmake

  Add the installation prefix of "cmocka" to CMAKE_PREFIX_PATH or set
  "cmocka_DIR" to a directory containing one of the above files.  If "cmocka"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  src/tests/CMakeLists.txt:26 (find_package)


-- Found PythonInterp: /usr/bin/python2.7 (found suitable exact version "2.7.15") 
-- Found GnuPG: /usr/bin/gpg (found suitable version "2.2.4", minimum required is "2.2") found components:  gpg gpgconf 
-- Configuring done
CMake Error at src/tests/CMakeLists.txt:29 (add_executable):
  Target "rnp_tests" links to target "cmocka::cmocka" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
```
The agent takes the last tag with the following script:
```
      last_committed_tag="$(git tag -l --sort=-v:refname | head -1)"
      VERSION=${last_committed_tag}
      git fetch
      git checkout "${VERSION}"
```
There isn't such error with HEAD. I suggest to add another tag.

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -17,7 +17,7 @@ jobs:
       run: snapcraft snap --output=rnp_latest.snap
     - name: Push snap
       env:
-        CHANNEL: edge
+        CHANNEL: beta
       run: |
         channel_version=$(snap info rnp | awk -v tag="$CHANNEL:" '$1 == tag { print $2 }')
         echo "The channel $CHANNEL contains version $channel_version."

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -20,7 +20,7 @@ jobs:
         CHANNEL: edge
       run: |
         channel_version=$(snap info rnp | awk -v tag="$CHANNEL:" '$1 == tag { print $2 }')
-        echo "The channel $CHANNEL contains version $snap_version."
+        echo "The channel $CHANNEL contains version $channel_version."
         snap_version=$(snap info rnp_latest.snap | awk '$1 == "version:" { print $2 }')
         if [ "${channel_version}" != "${snap_version}" ]; then
           echo "Pushing the new version $snap_version to channel $CHANNEL"

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -16,7 +16,17 @@ jobs:
         SNAPCRAFT_BUILD_ENVIRONMENT: host
       run: snapcraft snap --output=rnp_latest.snap
     - name: Push snap
+      env:
+        CHANNEL: edge
       run: |
-        echo "${{ secrets.SNAPCRAFT_RELEASE_LOGIN }}" > snap-exported-login
-        snapcraft login --with snap-exported-login
-        snapcraft push --release=edge rnp_latest.snap
+        channel_version=$(snap info rnp | awk -v tag="$CHANNEL:" '$1 == tag { print $2 }')
+        echo "The channel $CHANNEL contains version $snap_version."
+        snap_version=$(snap info rnp_latest.snap | awk '$1 == "version:" { print $2 }')
+        if [ "${channel_version}" != "${snap_version}" ]; then
+          echo "Pushing the new version $snap_version to channel $CHANNEL"
+          echo "${{ secrets.SNAPCRAFT_RELEASE_LOGIN }}" > snap-exported-login
+          snapcraft login --with snap-exported-login
+          snapcraft push --release=$CHANNEL rnp_latest.snap
+        else
+          echo "The version $snap_version already present in the channel. Cancelling the push."
+        fi

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -40,7 +40,8 @@ parts:
   botan2:
     plugin: make
     override-build: ./configure.py --prefix=/usr && snapcraftctl build
-    source: https://botan.randombit.net/releases/Botan-2.12.1.tar.xz
+    source: https://botan.randombit.net/releases/Botan-2.13.0.tar.xz
+    source-checksum: sha256/f57ae42a41e1091bca58f44f41addebd9a390b651603952c881ec89d50187e90
     build-packages:
       - g++
       - make

